File: java/com/lc/lc226.java
package lc;

import java.util.*;

public class lc226 {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return root;
        }
        invert(root);
        return root;
    }

    public void invert(TreeNode root) {
        if (root == null) {
            return;
        }
        TreeNode l = root.left;
        root.left = root.right;
        root.right = l;
        invert(root.left);
        invert(root.right);
    }
}

File: go/lc/lc226.go
package main

func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return root
	}
	queue := make([]*TreeNode, 0)
	queue = append(queue, root)
	for len(queue) > 0 {
		now := queue[0]
		queue = queue[1:]
		t := now.Left
		now.Left = now.Right
		now.Right = t
		if now.Left != nil {
			queue = append(queue, now.Left)
		}
		if now.Right != nil {
			queue = append(queue, now.Right)
		}
	}
	return root
}
